app/src/main/java/com/github/geohunt/app/model/database/firebase/FirebaseDatabase.kt

```diff
@@ -297,6 +300,69 @@ class FirebaseDatabase(activity: Activity) : Database {
     override suspend fun unfollow(follower: String, followee: String) {
         doFollow(follower, followee, follow = false)
     }
-}
 
+    /**
+     * Returns a list of users that liked a given challenge
+     * @param cid the challenge id
+     * @return a list of users that liked the challenge
+     */
+    internal fun getLikeRefById(cid: String): LazyRef<User> {
+        return object : BaseLazyRef<User>() {
+            override fun fetchValue(): Task<User> {
+                return dbChallengeRef.child(cid).child("likes").get().thenMap {
+                    val uid = it.key!!
+                    getUserById(uid).fetch().result
+                }
+            }
+
+            override val id: String
+                get() = cid
+        }
+    }
+
+    /**
+     * Insert a like for a user into the Firebase
+     * @param uid the user id
+     * @param cid the challenge id
+     * @return a task that will complete when the like is inserted
+     */
+    override fun insertUserLike(uid: String, cid: String): Task<Void> {
+        //Add the challenge to the user's liked challenges
+        return Tasks.whenAll(
+            dbUserRef.child(uid).child("likes").child(cid).setValue(true),
+            dbChallengeRef.child(cid).child("likes").child(uid).setValue(true)
```

Comment:
Incrementing number of likes. You should check the `doFollow` written by @Dario method for an idea of how to implement. Also you should mention #88

Reply:
Thank you for pointing it out, I have changed the implementation of the insertUserLike and removeUserLike following your comment.